### PR TITLE
Fix linting and TypeScript errors in the CDapp

### LINKED ISSUE #438: Fix linting and TypeScript errors in the CDapp

### PR DESCRIPTION
**Pull request by jakubcolony**
## Description

This PR attempts to free the CDapp from linting and TS errors. I aimed to remove as much dead code as possible, with the idea that it can be brought from the Dapp if needed. Upfront apologies for the size of the PR. I tried to group the commits logically as much as possible, so you might find it easier to review commit-by-commit. I did my best not to introduce any changes other than just fixing the errors. 

## Testing

There's few areas of the code this PR doesn't change, so testing should focus on ensuring no regression to the existing features was introduced.

Resolves #438

### CODE REVIEW

**Comment by jakubcolony** [comment]
The CDapp is now effectively error free, but there's something weird going with imports:
image

The property `WrongColony` is being read from a GQL generated type (so it's definitely not `undefined`) and my suspicions point towards a circular dependency. @willm30 I'll be probably coming your way to talk about these as I know you've dealt with them in the past

**Comment by willm30** [comment]
@jakubcolony Sure, no prob. One way to debug is to go through the stack trace and look for places where ColonyActionType is being imported before it gets re-exported from `~types/graphql.ts`

File: src/components/shared/OmniPicker/OmniPicker.tsx [hunk 1]
```diff
@@ -129,55 +129,67 @@ class OmniPicker extends Component<Props> {
     }
   }
 
+  /**
+   * @NOTE All of the following methods are actually used by the wrapped component,
+   * eslint just doesn't seem to be able to pick it up
+   */
+  // eslint-disable-next-line react/no-unused-class-component-methods
```
Comment (willm30) [inline src/components/shared/OmniPicker/OmniPicker.tsx:136]: Could turn this off for the entire block instead of multiple next-lines?

**Review by willm30 — APPROVED** [review]
Great job Jakub!

Didn't notice anything regressions while clicking around the app.

File: src/components/common/Dialogs/PermissionManagementDialog/helpers.ts [hunk 2]
```diff
@@ -110,7 +110,9 @@ export const useCanRoleBeSet = (
     Id.RootDomain,
   ]);
 
-  const rootAccounts = useTransformer(getAllRootAccounts, [colony]);
+  const rootAccounts = useTransformer(getAllRootAccounts, [
+    /* colony */
+  ]);
```
[the remark this answers is not in the document]
Comment (jakubcolony) [reply src/components/common/Dialogs/PermissionManagementDialog/helpers.ts:115]: The `getAllRootAccounts` transformer is temporarily not accepting the colony argument:
image

File: src/components/shared/Button/__tests__/Button.test.tsx
[the remark this answers is not in the document]
Comment (jakubcolony) [reply src/components/shared/Button/__tests__/Button.test.tsx:1]: Both really. But anything can be ported back from the Dapp if needed

File: src/components/common/Dialogs/PermissionManagementDialog/helpers.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (rdig) [reply src/components/common/Dialogs/PermissionManagementDialog/helpers.ts:115]: I'm redoing all these. Don't worry about them

File: src/components/shared/OmniPicker/OmniPicker.tsx (hunk 1 above)
Comment (rdig) [reply src/components/shared/OmniPicker/OmniPicker.tsx:136]: It can even be disabled for the whole file

**Review by rdig — APPROVED** [review]
Massive amounts of work here. Nice job! :100:

File: src/redux/sagas/setupUserContext.ts [hunk 3]
```diff
@@ -6,7 +6,6 @@ import colonySagas, { colonyCreateSaga } from './colony';
 import extensionSagas from './extensions';
 // import motionSagas from './motions';
 // import whitelistSagas from './whitelist';
-// import vestingSagas from './vesting';
```
[the remark this answers is not in the document]
Comment (jakubcolony) [reply src/redux/sagas/setupUserContext.ts:9]: For some reason I thought otherwise, brought it back

**Pull request merged**